### PR TITLE
Refactor execution modes

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: gears/src/baseapp/mod.rs [hunk 1]
```diff
@@ -195,7 +193,10 @@ impl<PSK: ParamsSubspaceKey, H: ABCIHandler, AI: ApplicationInfo> BaseApp<PSK, H
         ctx.block_gas_meter
             .consume_gas(gas_used, BLOCK_GAS_DESCRIPTOR)?;
 
-        MD::commit(ctx, &mut multi_store);
+        std::mem::drop(ctx);
```
Comment (dmvict) [inline gears/src/baseapp/mod.rs:196]: Just a question. 
Why do you need to drop it here?

**Review by dmvict — APPROVED** [review]

File: gears/src/baseapp/mod.rs (hunk 1 above)
Comment (NYBACHOK) [reply gears/src/baseapp/mod.rs:196]: To release mutable reference of `MultiBank<DB, AH::StoreKey, TransactionStore>` from ctx

**Comment by NYBACHOK** [comment]
@joneskm resolved conflicts

**Comment by NYBACHOK** [comment]
@joneskm resolved issue again

**Pull request merged**